File: backend/app/core/card_deck.py
"""Suit colors for playing cards. OG preview always uses four-color."""

from app.models.enums import CardDeck

DEFAULT_CARD_DECK = CardDeck.FOUR_COLOR

SUIT_BLACK = "#1A1710"
SUIT_RED = "#C7382F"
SUIT_DIAMOND = "#2E6FC4"
SUIT_CLUB = "#1F8A4C"

SUIT_COLORS: dict[CardDeck, dict[str, str]] = {
    CardDeck.CLASSIC: {
        "s": SUIT_BLACK,
        "h": SUIT_RED,
        "d": SUIT_RED,
        "c": SUIT_BLACK,
    },
    CardDeck.FOUR_COLOR: {
        "s": SUIT_BLACK,
        "h": SUIT_RED,
        "d": SUIT_DIAMOND,
        "c": SUIT_CLUB,
    },
}


def suit_color(suit: str, scheme: CardDeck = DEFAULT_CARD_DECK) -> str:
    palette = SUIT_COLORS[scheme]
    return palette.get(suit, SUIT_BLACK)


def parse_card_deck(value: str | CardDeck | None) -> CardDeck:
    if isinstance(value, CardDeck):
        return value
    try:
        return CardDeck(value)
    except (TypeError, ValueError):
        return DEFAULT_CARD_DECK

File: frontend/src/features/hands/lib/cardDeck.ts
import type { CardDeck } from "@/api/types/auth";

export type { CardDeck };

export const DEFAULT_CARD_DECK: CardDeck = "four_color";

export const CARD_FACE = "#FBF8F0";
export const SUIT_BLACK = "#1A1710";
export const SUIT_RED = "#C7382F";
export const SUIT_DIAMOND = "#2E6FC4";
export const SUIT_CLUB = "#1F8A4C";

const PALETTE: Record<CardDeck, Record<string, string>> = {
  classic: { s: SUIT_BLACK, h: SUIT_RED, d: SUIT_RED, c: SUIT_BLACK },
  four_color: { s: SUIT_BLACK, h: SUIT_RED, d: SUIT_DIAMOND, c: SUIT_CLUB },
};

export const CARD_DECK_LABELS: Record<CardDeck, string> = {
  classic: "Классическая",
  four_color: "Четырёхцветная",
};

export function isCardDeck(value: unknown): value is CardDeck {
  return value === "classic" || value === "four_color";
}

export function parseCardDeck(value: unknown): CardDeck {
  return isCardDeck(value) ? value : DEFAULT_CARD_DECK;
}

/** Suit ink on a cream card face. Same hex in both app themes. */
export function suitColor(suit: string, scheme: CardDeck = DEFAULT_CARD_DECK): string {
  return PALETTE[scheme][suit] ?? SUIT_BLACK;
}

export function isBlackSuit(suit: string, scheme: CardDeck = DEFAULT_CARD_DECK): boolean {
  return suitColor(suit, scheme) === SUIT_BLACK;
}
